addons/crm/wizard/crm_lead_to_opportunity.py

```diff
@@ -81,79 +81,63 @@ def _onchange_user(self):
                 values = self.env['crm.lead']._onchange_user_values(self.user_id.id if self.user_id else False)
                 self.team_id = values.get('team_id', False)
 
-    # NOTE JEM : is it the good place to test this ?
     @api.model
     def view_init(self, fields):
+        # JEM TDE FIXME: clean that brol
         """ Check some preconditions before the wizard executes. """
         for lead in self.env['crm.lead'].browse(self._context.get('active_ids', [])):
             if lead.probability == 100:
                 raise UserError(_("Closed/Dead leads cannot be converted into opportunities."))
         return False
 
-    def _convert_opportunity(self, vals):
+    def action_apply(self):
+        if self.name == 'merge':
+            result_opportunity = self._action_merge()
+        else:
+            result_opportunity = self._action_convert()
+
+        return result_opportunity.redirect_lead_opportunity_view()
+
+    def _action_merge(self):
+        result_opportunity = self.with_context(active_test=False).opportunity_ids.merge_opportunity()
+        if not result_opportunity.active:
+            result_opportunity.write({'active': True, 'activity_type_id': False, 'lost_reason': False})
+
+        if result_opportunity.type == "lead":
+            self._convert_and_allocate(result_opportunity, [self.user_id.id], team_id=self.team_id.id)
+        elif not self._context.get('no_force_assignation') or not result_opportunity.user_id:
+            result_opportunity.write({
+                'user_id': self.user_id.id,
+                'team_id': self.team_id.id,
+            })
```

Comment:
while we are at it, could we instead prepare a value dict and write everything at the end if needed ?

Reply:
Actually I have another task which will remove the first write and call action_unarchive instead :D . So there won't be two consecutive write anymore.